### PR TITLE
Upgrade to .NET 7 (TKX-18)

### PR DESCRIPTION
**Pull request by joaofbvieira**

### CODE REVIEW

**Review by emanuelgaspar — APPROVED** [review]

File: src/Trakx.Circle.ApiClient/CircleApiConfiguration.cs
Comment (emanuelgaspar) [inline src/Trakx.Circle.ApiClient/CircleApiConfiguration.cs:1]: I recommend setting `AllowGlobal = true` in the AWS parameter.

Comment (emanuelgaspar) [reply src/Trakx.Circle.ApiClient/CircleApiConfiguration.cs:1]: Plus, let's get rid of the `#nullable disable` hack and do our standard "nullable friendly" approach.

Comment (joaofbvieira) [reply src/Trakx.Circle.ApiClient/CircleApiConfiguration.cs:1]: I'll create a new task to check this later. I tried to focus mostly in the .NET migration tbh. Some stuff could be improved for sure.
Task: TKX955.

File: src/Trakx.Circle.ApiClient/ClientRegistrations.tt
[the remark this answers is not in the document]
Comment (joaofbvieira) [reply src/Trakx.Circle.ApiClient/ClientRegistrations.tt:1]: Created task TKX956.

**Review by emanuelgaspar — APPROVED** [review]

**Pull request merged**